aten/src/ATen/cuda/Exceptions.h

```diff
@@ -70,21 +71,28 @@ namespace at { namespace cuda { namespace solver {
 C10_EXPORT const char* cusolverGetErrorMessage(cusolverStatus_t status);
 }}} // namespace at::cuda::solver
 
-#define TORCH_CUSOLVER_CHECK(EXPR)                                              \
-  do {                                                                          \
-    cusolverStatus_t __err = EXPR;                                              \
-    if (__err == CUSOLVER_STATUS_EXECUTION_FAILED) {                            \
-      TORCH_CHECK(__err == CUSOLVER_STATUS_SUCCESS,                             \
-                  "cusolver error: ",                                           \
-                  at::cuda::solver::cusolverGetErrorMessage(__err),             \
-                  ", when calling `" #EXPR "`",                                 \
-                  ". This error may appear if the input matrix contains NaN."); \
-    } else {                                                                    \
-      TORCH_CHECK(__err == CUSOLVER_STATUS_SUCCESS,                             \
-                  "cusolver error: ",                                           \
-                  at::cuda::solver::cusolverGetErrorMessage(__err),             \
-                  ", when calling `" #EXPR "`");                                \
-    }                                                                           \
+// When cuda < 11.5, cusolver raises CUSOLVER_STATUS_EXECUTION_FAILED when input contains nan.
+// When cuda >= 11.5, cusolver normally finishes execution and sets info array indicating convergence issue.
```

Comment:
Is this documented in cusolver's docs? Is there a list of functions that follow the described behavior? Is it applicable to all functions in cusolver?

Reply:
This is taken from this comment here: https://github.com/pytorch/pytorch/blob/master/test/test_linalg.py#L2940.

Given that the other tests also raise LinAlgError after this change, it seems that it should be true for all the functions. I could not find anything explicitly written in the docs.